client/app/pages/queries/view.js

```diff
@@ -523,6 +523,21 @@ function QueryViewCtrl(
     },
   );
 
+  if (!$scope.query.isNew()) {
+    $scope.$watch(
+      'query.options && query.options.parameters',
+      (parameters) => {
+        const params = extend({}, $location.search());
+        parameters.forEach((param) => {
+          extend(params, param.toUrlParams());
+        });
+        Object.keys(params).forEach(key => params[key] == null && delete params[key]);
+        $location.search(params);
```

Comment:
I initially thought this logic was only for the query page, but it's actually to all pages that contain parameters.

Will need to find a replacement to that. The fix for #4004 can probably be done while doing it.